### ISSUE TITLE
Crashes after a few seconds when libpam-tmpdir is installed

### ISSUE THREAD

**Issue by kkeijzer** [flathub/com.spotify.Client#84]
```
kevin@vanadium:~$ flatpak run com.spotify.Client
/app/extra/bin/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/bin/spotify)
Gtk-Message: 22:36:47.564: Failed to load module "canberra-gtk-module"
/app/extra/share/spotify/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/share/spotify/spotify)
/proc/self/exe: /app/lib/libcurl-gnutls.so.4: no version information available (required by /proc/self/exe)
kevin@vanadium:~$ terminating with uncaught exception of type boost::filesystem::filesystem_error: boost::filesystem::temp_directory_path: Not a directory: "/tmp/user/1000"
```

I have `libpam-tmpdir` installed, which probably sets the `TMPDIR` to `/tmp/user/1000`. I'd rather not uninstall it, because I prefer `/tmp` to be separated between users.

Changing the `TMPDIR` variable, however, does not help:

```
kevin@vanadium:~$ export TMPDIR=/tmp
kevin@vanadium:~$ echo $TMPDIR
/tmp
kevin@vanadium:~$ flatpak run com.spotify.Client
/app/extra/bin/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/bin/spotify)
Gtk-Message: 22:39:40.129: Failed to load module "canberra-gtk-module"
/app/extra/share/spotify/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/share/spotify/spotify)
/proc/self/exe: /app/lib/libcurl-gnutls.so.4: no version information available (required by /proc/self/exe)
kevin@vanadium:~$ terminating with uncaught exception of type boost::filesystem::filesystem_error: boost::filesystem::temp_directory_path: Not a directory: "/tmp/user/1000"
```
Nor does unsetting it:

```
kevin@vanadium:~$ export TMPDIR=
kevin@vanadium:~$ echo $TMPDIR 

kevin@vanadium:~$ flatpak run com.spotify.Client
/app/extra/bin/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/bin/spotify)
Gtk-Message: 22:40:39.462: Failed to load module "canberra-gtk-module"
/app/extra/share/spotify/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/share/spotify/spotify)
/proc/self/exe: /app/lib/libcurl-gnutls.so.4: no version information available (required by /proc/self/exe)
kevin@vanadium:~$ terminating with uncaught exception of type boost::filesystem::filesystem_error: boost::filesystem::temp_directory_path: Not a directory: "/tmp/user/1000"
```

_Originally posted by @kkeijzer in https://github.com/flathub/com.spotify.Client/issues/48#issuecomment-501447372_

**Comment by TingPing** [comment]
The package already unsets TMPDIR.. https://github.com/flathub/com.spotify.Client/commit/2d55ed957dfe74739186fcb0c8e22b24fb098ed1

Does `flatpak run --command=sh com.spotify.Client` then `echo $TMPDIR` show a value?

**Comment by kkeijzer** [comment]
No, it does not,

```
kevin@vanadium:~$ flatpak run --command=sh com.spotify.Client
sh-4.4$ echo $TMPDIR

sh-4.4$
```

I know `$TMPDIR` should be unset, but for some reason this does not work with `libpam-tmpdir` enabled.

```
kevin@vanadium:~$ flatpak run com.spotify.Client
/app/extra/bin/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/bin/spotify)
Gtk-Message: 20:04:42.676: Failed to load module "canberra-gtk-module"
/app/extra/share/spotify/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/share/spotify/spotify)
/proc/self/exe: /app/lib/libcurl-gnutls.so.4: no version information available (required by /proc/self/exe)
kevin@vanadium:~$ terminating with uncaught exception of type boost::filesystem::filesystem_error: boost::filesystem::temp_directory_path: Not a directory: "/tmp/user/1000"
```

**Comment by kkeijzer** [comment]
Ok, this changes the error:

```
kevin@vanadium:~$ export TMP=

kevin@vanadium:~$ flatpak run com.spotify.Client 
/app/extra/bin/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/bin/spotify)
Gtk-Message: 22:05:41.573: Failed to load module "canberra-gtk-module"
/app/extra/share/spotify/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/share/spotify/spotify)
/proc/self/exe: /app/lib/libcurl-gnutls.so.4: no version information available (required by /proc/self/exe)
kevin@vanadium:~$ terminating with uncaught exception of type boost::filesystem::filesystem_error: boost::filesystem::temp_directory_path: Not a directory
```

**Comment by kkeijzer** [comment]
And running this makes it work even with `libpam-tmpdir` installed:

```
kevin@vanadium:~$ export TMP=/tmp

kevin@vanadium:~$ flatpak run com.spotify.Client 
/app/extra/bin/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/bin/spotify)
Gtk-Message: 22:05:41.573: Failed to load module "canberra-gtk-module"
/app/extra/share/spotify/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/share/spotify/spotify)
/proc/self/exe: /app/lib/libcurl-gnutls.so.4: no version information available (required by /proc/self/exe)
```

I'm unsure what would be the best `TMP` variable though, because this still selects the actual `/tmp`, which is an antipattern when using `libpam-tmpdir`.

**Comment by TingPing** [comment]
Hmm, does `flatpak run --env=TMPDIR=/tmp com.spotify.Client` work when `TMP` on host is unset?

**Comment by kkeijzer** [comment]
I made an override, and now it works for me:

```
kevin@vanadium:~$ cat /var/lib/flatpak/overrides/com.spotify.Client 
[Environment]
TMP=/tmp
```

It doesn't matter what the `TMP` variable on the host is. (In my case, it's `/tmp/user/1000`.)

**Comment by TingPing** [comment]
This is probably a discussion that should still be had against `flatpak` itself. Since it has a private `/tmp` it should not be allowing host state to mess with that.

**Comment by Cherkah** [comment]
same issue as previous.
libpam-tmpdir = installed

after logging, i've to open spotify three times to remain it works. if do not, the app crash!

/app/extra/bin/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/bin/spotify) /app/extra/share/spotify/spotify: /app/lib/libcurl-gnutls.so.4: no version information available (required by /app/extra/share/spotify/spotify) /proc/self/exe: /app/lib/libcurl-gnutls.so.4: no version information available (required by /proc/self/exe) [0721/231300.002769:ERROR:buffer_manager.cc(488)] [.DisplayCompositor]GL ERROR :GL_INVALID_OPERATION : glBufferData: <- error from previous GL command terminating with uncaught exception of type boost::filesystem::filesystem_error: boost::filesystem::temp_directory_path: N'est pas un dossier: "/tmp/user/1000"

**Comment by TingPing** [comment]
I can only repeat that maybe this should be reported against `flatpak` so many times.
